src/hotspot/share/oops/resolvedFieldEntry.hpp

```diff
@@ -55,6 +55,17 @@ class ResolvedFieldEntry {
   u1 _flags;                    // Flags: [0000|00|is_final|is_volatile]
   u1 _get_code, _put_code;      // Get and Put bytecodes of the field
 
+  void copy_from(const ResolvedFieldEntry& other) {
+    _field_holder = other._field_holder;
+    _field_offset = other._field_offset;
+    _field_index = other._field_index;
+    _cpool_index = other._cpool_index;
+    _tos_state = other._tos_state;
+    _flags = other._flags;
+    _get_code = other._get_code;
```

Comment:
The fields `_get_code` and `_put_code` are normally set atomically, does this need to be the case when copying as well?

Reply:
This is done inside while the ResolvedFieldEntries are being prepared during class rewriting. All access is single threaded so there's no need for atomic operations.